src/coordinatorapi.cpp

```diff
@@ -658,16 +658,25 @@ void CoordinatorAPI::sendCkptFilename()
   // Tell coordinator to record our filename in the restart script
   string ckptFilename = ProcessInfo::instance().getCkptFilename();
   string hostname = jalib::Filesystem::GetCurrentHostname();
-  JTRACE("recording filenames") (ckptFilename) (hostname);
   DmtcpMessage msg;
   if (dmtcp_unique_ckpt_enabled && dmtcp_unique_ckpt_enabled()) {
     msg.type = DMT_UNIQUE_CKPT_FILENAME;
   } else {
     msg.type = DMT_CKPT_FILENAME;
   }
-  msg.extraBytes = ckptFilename.length() + 1 + hostname.length() + 1;
+  // Tell coordinator type of remote shell command used ssh/rsh
+  string shellType("");
+  const char *remoteShellType = getenv(ENV_VAR_REMOTE_SHELL_CMD);
+  if(remoteShellType != NULL) {
+    shellType = remoteShellType;
```

Comment:
Will this create a memory leak?  We're converting a C string to a new string object.  shellType used to point to a string object, "", and now points to a different string object.  How do we know that these two string objects are both freed?

I'd have to check up on the rules of C++ to be sure what happens here.  But if it's possibly to make shellType a `char *` instead of a `string`, then anybody reading this code will immediately know that it doesn't leak memory.